sql/logic_test.go

```diff
@@ -172,6 +172,20 @@ func (t *logicTest) close() {
 
 // setUser sets the DB client to the specified user, creating it if needed.
 func (t *logicTest) setUser(user string) {
+	if t.db != nil {
+		var dbName string
+
+		if err := t.db.QueryRow("SHOW DATABASE").Scan(&dbName); err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
```

Comment:
Should add a comment about what this is doing: `When switching to a new user, propagate the existing DATABASE setting to the new connection`.